app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -75,6 +76,20 @@ def validate_authorize_form
       end
     end
 
+    def force_login_if_prompt_param_is_login_and_request_is_external
+      return unless user_signed_in? && @authorize_form.prompt == 'login'
+      sign_out unless referring_host.to_s.casecmp(Figaro.env.domain_name).zero?
+    end
+
+    def referring_host
+      begin
+        ret = URI.parse(request.referer).host
+      rescue URI::InvalidURIError
+        ret = nil
+      end
+      ret
+    end
```

Comment:
Could this be simplified to this?
```ruby
def referring_host(referer)
  URI.parse(referer).host
rescue URI::InvalidURIError
  nil
end
```
I think this is something Rubocop would flag, but Code Climate does not seem to be running Rubocop. Do you have a Rubocop plugin for your editor?

Reply:
Ah I like

Reply:
originally I just had a blank rescue too but I got slapped on the wrist

Reply:
rubocop didn't flag tho.  i run it on all my checkins

Reply:
Is there a particular reason for using `casecmp` and not simply comparing the referring_host with Figaro.env.domain_name? Is there ever a case where the internal referring host would contain capital letters?

Can we add a test for when `request.referer` is `nil` or a value that URI can't parse?

Reply:
I was worried about the external referring host containing capitals

Reply:
Why do we care if the external host has capitals? Is it ever possible for an external referring host to match our domain name?

Reply:
Ah you are correct.  Now that I'm not using Figaro it's no longer an issue.

Reply:
That said the code is more flexible when when insensitively comparing a referer which can be capitalized.  It just so happens in this specific case it is not significant.  I was going to take that case compare and make it a method that made it clear what the intent was since the case compare is so obtuse looking.